src/zcash/Note.hpp

```diff
@@ -116,6 +116,82 @@ class SproutNotePlaintext : public BaseNotePlaintext {
                                         ) const;
 };
 
+typedef std::pair<SaplingEncCiphertext, SaplingNoteEncryption> SaplingNotePlaintextEncryptionResult;
+
+class SaplingNotePlaintext : public BaseNotePlaintext {
+public:
+    diversifier_t d;
+    uint256 rcm;
+
+    SaplingNotePlaintext() {}
+
+    SaplingNotePlaintext(const SaplingNote& note, std::array<unsigned char, ZC_MEMO_SIZE> memo);
```

Comment:
We should add some comments to help understand the receiver vs sender roles; e.g. add here 'Enables a note recipient to decrypt their note'